### PR TITLE
MinioAdmin: Add {add,update,delete,list,get}_service_account APIs

### LINKED ISSUE #1307: Request to add user svcacct feature to minio-py
The access keys section displays all Access Keys associated to the authenticated user.

I found that minio-py lacks support for mc admin user svcacct. I searched through all the code and didn't find any mention of svcacct. This is surprising. Perhaps other users also need this. I need to complete the code for this module in the next few days. If needed, I can submit these codes to this repository

[User Management](https://min.io/docs/minio/linux/administration/identity-access-management/minio-user-management.html)

### PR DESCRIPTION
**Pull request by Alveel**
Added the missing functions to manage service accounts with this library.

This closes #1307.

### CODE REVIEW

**Comment by Alveel** [comment]
Ready for review!

**Comment by Alveel** [comment]
@balamurugana did you have a chance to take a peek?

**Comment by Alveel** [comment]
Some of it is not the prettiest, I welcome suggestions and improvements.

File: minio/minioadmin.py [hunk 1]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
             query_params={"bucket": bucket}
         )
         return response.data.decode()
+
+    def service_account_info(self, access_key):
```
Comment (balamurugana) [inline minio/minioadmin.py:721]: 1. Add typing
2. The method can be renamed as `get_service_account_info`

File: minio/minioadmin.py [hunk 2]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+            _COMMAND.SERVICE_ACCOUNT_INFO,
+            query_params={"accessKey": access_key},
+            preload_content=False,
+        )
+        plain_data = decrypt(
+            response, self._provider.retrieve().secret_key,
+        )
+        return plain_data.decode()
+
+    def service_account_list(self, user: str):
```
Comment (balamurugana) [inline minio/minioadmin.py:734]: 1. Add typing
2. The method can be renamed to `list_service_account`

File: minio/minioadmin.py [hunk 3]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+            preload_content=False,
+        )
+        plain_data = decrypt(
+            response, self._provider.retrieve().secret_key,
+        )
+        return plain_data.decode()
+
+    def service_account_add(self,
+                            access_key: str = "",
+                            secret_key: str = "",
```
Comment (balamurugana) [inline minio/minioadmin.py:749]: 1. access/secret key args should be `access_key: str | None = None` and `secret_key: str | None = None`
2. Add typing to return value

File: minio/minioadmin.py [hunk 4]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+                            description: str | None = None,
+                            policy_file: str | None = None,
+                            expiration: str | None = None,
+                            status: str | None = None):
+        """
+        Add a new service account with the given access key and secret key
+        """
+        if ((secret_key and access_key == "")
+                or (access_key and secret_key == "")):
+            raise ValueError("access_key must be provided if secret_key is "
```
Comment (balamurugana) [inline minio/minioadmin.py:760]: Just do
```py
if (access_key is None) ^ (secret_key is None):
    raise ValueError("both access key and secret key must be provided")
elif access_key == "" or secret_key == "":
    raise ValueError("access key or access key must not be empty")
```

File: minio/minioadmin.py [hunk 5]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+            query_params={"accessKey": access_key},
+            body=encrypt(body, self._provider.retrieve().secret_key),
+            preload_content=False,
+        )
+        plain_data = decrypt(
+            response, self._provider.retrieve().secret_key,
+        )
+        return plain_data.decode()
+
+    def service_account_update(self,
```
Comment (balamurugana) [inline minio/minioadmin.py:792]: Rename this method as `update_service_account`

File: minio/minioadmin.py [hunk 6]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+                               access_key: str,
+                               secret_key: str | None = None,
+                               name: str | None = None,
+                               description: str | None = None,
+                               policy_file: str | None = None,
+                               expiration: str | None = None,
+                               status: str | None = None) -> str:
```
Comment (balamurugana) [inline minio/minioadmin.py:799]: Add return type

File: minio/minioadmin.py [hunk 7]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+        body = json.dumps(data).encode()
+        response = self._url_open(
+            "POST",
+            _COMMAND.SERVICE_ACCOUNT_UPDATE,
+            query_params={"accessKey": access_key},
+            body=encrypt(body, self._provider.retrieve().secret_key),
+        )
+        return response.data.decode()
+
+    def service_account_delete(self, access_key) -> str:
```
Comment (balamurugana) [inline minio/minioadmin.py:830]: 1. add parameter typing
2. rename this method as `delete_service_account`

**Review by balamurugana — CHANGES_REQUESTED** [review]

File: minio/minioadmin.py (hunk 1 above)
Comment (balamurugana) [inline minio/minioadmin.py:721]:
```suggestion
    def get_service_account_info(self, access_key: str) -> str:
```

File: minio/minioadmin.py (hunk 2 above)
Comment (balamurugana) [inline minio/minioadmin.py:734]:
```suggestion
    def list_service_account(self, user: str) -> str:
```

File: minio/minioadmin.py (hunk 3 above)
Comment (balamurugana) [inline minio/minioadmin.py:749]:
```suggestion
                            access_key: str | None = None,
                            secret_key: str | None = None,
```

File: minio/minioadmin.py [hunk 8]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+            _COMMAND.SERVICE_ACCOUNT_LIST,
+            query_params={"user": user},
+            preload_content=False,
+        )
+        plain_data = decrypt(
+            response, self._provider.retrieve().secret_key,
+        )
+        return plain_data.decode()
+
+    def service_account_add(self,
```
Comment (balamurugana) [inline minio/minioadmin.py:747]:
```suggestion
    def add_service_account(self,
```

File: minio/minioadmin.py [hunk 9]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+                            policy_file: str | None = None,
+                            expiration: str | None = None,
+                            status: str | None = None):
+        """
+        Add a new service account with the given access key and secret key
+        """
+        if ((secret_key and access_key == "")
+                or (access_key and secret_key == "")):
+            raise ValueError("access_key must be provided if secret_key is "
+                             "given and vice versa")
```
Comment (balamurugana) [inline minio/minioadmin.py:761]:
```suggestion
        if (access_key is None) ^ (secret_key is None):
            raise ValueError("both access key and secret key must be provided")
        elif access_key == "" or secret_key == "":
            raise ValueError("access key or access key must not be empty")
```

File: minio/minioadmin.py (hunk 5 above)
Comment (balamurugana) [inline minio/minioadmin.py:792]:
```suggestion
    def update_service_account(self,
```

File: minio/minioadmin.py (hunk 7 above)
Comment (balamurugana) [inline minio/minioadmin.py:830]:
```suggestion
    def delete_service_account(self, access_key: str) -> str:
```

File: minio/minioadmin.py [hunk 10]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+        return plain_data.decode()
+
+    def add_service_account(self,
+                            access_key: str | None = None,
+                            secret_key: str | None = None,
+                            name: str | None = None,
+                            description: str | None = None,
+                            policy_file: str | None = None,
+                            expiration: str | None = None,
+                            status: str | None = None):
```
Comment (balamurugana) [inline minio/minioadmin.py:754]:
```suggestion
                            status: str | None = None) -> str:
```

File: minio/minioadmin.py [hunk 11]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+                            description: str | None = None,
+                            policy_file: str | None = None,
+                            expiration: str | None = None,
+                            status: str | None = None) -> str:
+        """
+        Add a new service account with the given access key and secret key
+        """
+        if (access_key is None) ^ (secret_key is None):
+            raise ValueError("both access key and secret key must be provided")
+        elif access_key == "" or secret_key == "":
```
Comment (balamurugana) [inline minio/minioadmin.py:760]:
```suggestion
        if access_key == "" or secret_key == "":
```

File: minio/minioadmin.py [hunk 12]
```diff
@@ -712,3 +717,121 @@ def bucket_quota_get(self, bucket: str) -> str:
...
+        if expiration:
+            data["expiration"] = expiration
+        if status:
+            data["status"] = status
+
+        body = json.dumps(data).encode()
+        response = self._url_open(
+            "PUT",
+            _COMMAND.SERVICE_ACCOUNT_ADD,
+            query_params={"accessKey": access_key},
```
Comment (balamurugana) [inline minio/minioadmin.py:783]:
```suggestion
```

File: minio/minioadmin.py [hunk 13]
```diff
@@ -712,3 +717,120 @@ def bucket_quota_get(self, bucket: str) -> str:
             query_params={"bucket": bucket}
         )
         return response.data.decode()
+
+    def get_service_account_info(self, access_key: str) -> str:
```
Comment (balamurugana) [inline minio/minioadmin.py:721]:
```suggestion
    def get_service_account(self, access_key: str) -> str:
```

**Review by balamurugana — APPROVED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**